app/scripts/lib/transaction/metrics.ts

```diff
@@ -973,6 +984,12 @@ async function buildEventFragmentProperties({
   const swapAndSendMetricsProperties =
     getSwapAndSendMetricsProps(transactionMeta);
 
+  // Add Entropy Properties
+  const hdEntropyIndex = transactionMetricsRequest.getHDEntropyIndex();
+  const hdEntropyProperties = {
+    hd_entropy_index: hdEntropyIndex === -1 ? undefined : hdEntropyIndex,
```

Comment:
We don't make this check in the other code branches, should we for consistency?

Reply:
Good call! I thought there is no option that we will pass -1 in any other cases, but I checked again and there is! I think we just need to update `getHDEntropyIndex()` and it should cover other code branches as well.